src/bootstrap/lib.rs

```diff
@@ -482,6 +482,12 @@ impl Build {
                   .env("RUSTC_DEBUGINFO_LINES", self.config.rust_debuginfo_lines.to_string());
         }
 
+        // Since the metadata of dependencies can conflict, we're adding a
+        // special suffix to make it unique.
+        if mode == Mode::Librustc {
```

Comment:
Yeah, I originally thought that libstd has absolutely zero dependencies, but turns out it's not. Fortunately, either the version is set to 0.0.0 or the feature set is different so the metadata differs. Anyway, I will enable it for other types too due to the potential risk of breaking it in future.